StreamChatClient_v3/Controllers/ChannelController.swift

```diff
@@ -5,21 +5,60 @@
 import CoreData
 import Foundation
 
+extension Client {
+    /// Creates a new `ChannelController` for the channel with the provided id and options.
+    ///
+    /// - Parameter channelId: The id of the channel this controller represents.
+    /// - Parameter options: Query options (See `QueryOptions`)
+    /// - Returns: A new instance of `ChannelController`.
+    ///
+    public func channelController(for channelId: ChannelId, options: QueryOptions = .all) -> ChannelControllerGeneric<ExtraData> {
+        .init(channelQuery: .init(channelId: channelId, options: options), client: self)
+    }
+    
+    /// Creates a new `ChannelController` for the channel with the provided id.
+    ///
+    /// - Parameter channelQuery: The ChannelQuery this controller represents
+    /// - Returns: A new instance of `ChannelController`.
+    ///
+    public func channelController(for channelQuery: ChannelQuery<ExtraData>) -> ChannelControllerGeneric<ExtraData> {
+        .init(channelQuery: channelQuery, client: self)
+    }
+}
+
+/// A convenience typealias for `ChannelControllerGeneric` with `DefaultDataTypes`
+public typealias ChannelController = ChannelControllerGeneric<DefaultDataTypes>
+
 /// `ChannelController` allows observing and mutating the controlled channel.
 ///
 ///  ... you can do this and that
 ///
-public class ChannelController<ExtraData: ExtraDataTypes>: Controller {
-    public let channelId: ChannelId
-    public weak var delegate: ChannelControllerDelegate?
+public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, DelegateCallbable {
+    /// The ChannelQuery this controller observes.
+    public let channelQuery: ChannelQuery<ExtraData>
     
-    private let viewContext: NSManagedObjectContext
-    private let worker: ChannelUpdater<ExtraData>
+    /// The `ChatClient` instance this controller belongs to.
+    public let client: Client<ExtraData>
+    
+    /// The channel matching the channelId. To observe updates to the channel, set your class as a delegate of this controller and call `startUpdating`.
+    public private(set) lazy var channel: ChannelModel<ExtraData>? = {
+        log.warning("Accessing `channels` before calling `startUpdating()` always results in nil channel.")
+        return nil
+    }()
```

Comment:
I think the controller also needs to expose `[MessageModel]`. This `channel` variable is just to get the details about the channel itself.